File: C/05_1Dim_array/BOJ_07_4344_평균은넘겠지.c
#include <stdio.h>
main() {
	int t, n;
	scanf("%d", &t);
	double* result = (double*)malloc(sizeof(double) * t);
	for (int i = 0; i < t; i++) {
		int score[1000];
		double sum = 0, cnt = 0;
		scanf("%d", &n);
		for (int j = 0; j < n; j++) {
			scanf("%d", &score[j]);
			sum += score[j];
		}
		double avg = (sum / n);
		for (int k = 0; k < n; k++) {
			if (score[k] > avg) {
				cnt++;
			}
		}
		result[i] = ((cnt / n) * 100);
	}
	for (int i = 0; i < t; i++) {
		printf("%.3lf%%\n", result[i]);
	}
}

File: JAVA/05_1Dim_array/BOJ_07_4344_평균은넘겠지.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int c = Integer.parseInt(br.readLine());
        double avg = 0;
        double cnt[] = new double[c];
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < c; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine(), " ");
            int n = Integer.parseInt(st.nextToken());
            int score[] = new int[n];
            for (int j = 0; j < n; j++) {
                score[j] = Integer.parseInt(st.nextToken());
                avg += score[j];
            }
            avg /= n;
            for (int k = 0; k < n; k++) {
                if (score[k] > avg) cnt[i]++;
            }
            sb.append(String.format("%.3f", cnt[i] / n * 100)).append('%').append('\n');
            avg = 0;
        }
        System.out.println(sb);
    }
}